### PR TITLE
diagnostics: unit tests to guard against message changes

### PR DESCRIPTION
**Pull request by sosiouxme**

### CODE REVIEW

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -378,7 +380,7 @@ func (m *Master) Start() error {
 	}
 
 	if m.api {
-		glog.Infof("Starting master on %s (%s)", m.config.ServingInfo.BindAddress, version.Get().String())
+		glog.Infof(MasterStartupMessage, m.config.ServingInfo.BindAddress, version.Get().String())
```
Comment (sosiouxme) [inline pkg/cmd/server/start/start_master.go:383]: @liggitt is this too invasive? Pretty much all the log messages diagnostics looks for are stuffed directly into glog so I can only get at them by pulling them out into exported constants (which assumes they're even in code we control).

However this is pretty effective for testing the relevance and correctness of the log matchers.

Comment (liggitt) [inline pkg/cmd/server/start/start_master.go:383]: hrmm, it makes it more painful to actually use the message and see the format string is being used correctly... not sure if `go vet` will catch format errors in referenced format strings like this

Comment (sosiouxme) [inline pkg/cmd/server/start/start_master.go:383]: Pretty sure `go vet` will catch it
https://code.google.com/p/go/source/detail?r=29665c4bae69
(they should of course be constants not vars)

But I agree it's awkward to impose this on the various places these
messages originate. OTOH... it's an extra signal to pay attention when
making changes to that log message.

Comment (liggitt) [inline pkg/cmd/server/start/start_master.go:383]: I think needing to do this is an indication that scraping log output for regex matches on human readable messages isn't a very solid way to do diagnostics... I don't think externalizing the format strings for that purpose is a great idea. I would rather share a constant message ID that we use in the master log message and the diagnostics recognizes

Comment (sosiouxme) [inline pkg/cmd/server/start/start_master.go:383]: I agree, it's shaky.

I could get on board with logging message IDs, but that would still leave us scraping logs for message IDs (which we hope are unique) and not having any good way to check that the message source actually emits that message ID (someone could look at it, wonder why it was there, remove/change it, and not suffer any failed tests). So I'm not sure it solves the problem any better, unless we externalize the message ID to enable the same kind of testing demonstrated in this PR.

We also may have no choice but scraping human-directed messages in some situations where they come from somewhere outside our influence and/or can't be changed.

You could argue that OpenShift shouldn't force us to look through human-directed server logs to figure out what is going on :)

Any ideas on where to go with this?

**Pull request closed without merging**